src/jmcomic_ai/core.py

```
@@ -302,51 +307,246 @@ def get_ranking(self, period: str = "day", page: int = 1) -> dict[str, Any]:
+                # Send detailed album info
+                import json
+                album_dict = {
+                    "id": str(album.album_id),
+                    "title": str(album.name),
+                    "author": str(album.author),
+                    "chapter_count": len(album),
+                    "tags": album.tags,
+                }
+                msg = f"📚 Album Info: {json.dumps(album_dict, ensure_ascii=False)}"
+                service_logger.info(msg)
+                service._safe_ctx_call(ctx, loop, lambda: ctx.info(msg), "Failed to send album info to ctx")
```

Ruff (0.14.13)

Undefined name `service`

(F821)